### PR TITLE
Add sliders to control organization name search

### PR DESCRIPTION
**Pull request by okkays**
https://github.com/usdoj-crt/crt-portal-management/issues/1705

## What does this change?

- 🌎 We can fuzzy search with ~name right now
- ⛔ The search isn't configurable, which means three letter search terms get the same fuzziness as long ones
- ✅ This commit adds sliders to control the "percentag" of fuzziness applied to the search

## Screenshots (for front-end PR):

Image

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/cts_forms/filters.py [hunk 1]
```diff
@@ -310,28 +336,36 @@ class OctetLength(Func):
     output_field = IntegerField()
 
 
-def filter_by_similar(queryset, column, search_text):
+def filter_by_similar(queryset, column, *, query, soundslike, lookslike):
```
Comment (kewitham) [inline crt_portal/cts_forms/filters.py:339]: I'm a little confused about how the "sounds like" and "looks like" variables impact each other. Like if I set the looks like filter to be too strict will I exclude some of the sounds like filtering? We should probably explain that a bit for the user on the front end too.

File: crt_portal/cts_forms/templates/forms/complaint_view/index/filters/incident_location_filter.html [hunk 2]
```diff
@@ -1,18 +1,11 @@
...
-{% endif %}
-  <div id="incident_location-filter">
-    <label for="id_{{form.fields.location_name.widget.attrs.name}}">
-      {{ form.location_name }}
-    </label>
-  </div>
-{% if ENABLED_FEATURES.fuzzy_location_name %}
-</span>
-{% endif %}
+<div id="incident_location-filter">
```
Comment (kewitham) [inline crt_portal/cts_forms/templates/forms/complaint_view/index/filters/incident_location_filter.html:8]: should we have some indication in the filter pills above the table that fuzziness is being used?

**Review by kewitham — COMMENTED** [review]
a couple questions but this is really cool!

**Comment by ilodidoj** [comment]
@okkays I have not had a chance to test this PR. Is the slider keyboard accessible? All input must be key board accessible for 508 and accessibility compliance.

**Comment by okkays** [comment]
@ilodidoj They are! I'd tested that myself, but they're also uswds components so we can trust they've done their due diligence there, too

**Comment by ilodidoj** [comment]
> @ilodidoj They are! I'd tested that myself, but they're also uswds components so we can trust they've done their due diligence there, too

Thanks for looking into it.  Trust but verify 😄

File: crt_portal/cts_forms/templates/forms/complaint_view/index/filters/incident_location_filter.html (hunk 2 above)
Comment (okkays) [reply crt_portal/cts_forms/templates/forms/complaint_view/index/filters/incident_location_filter.html:8]: hmm good point. To keep it simple, I might just add a suffix like "Organization name (fuzzy)" or "Organization name (exact)"

File: crt_portal/cts_forms/filters.py (hunk 1 above)
Comment (okkays) [reply crt_portal/cts_forms/filters.py:339]: they OR together, so they don't actually impact each other - if you add more sounds like, you get more sounds like. If you then adds more looks like, you get more looks like in the mix, too.

File: crt_portal/cts_forms/templates/forms/complaint_view/index/filters/incident_location_filter.html (hunk 2 above)
Comment (okkays) [reply crt_portal/cts_forms/templates/forms/complaint_view/index/filters/incident_location_filter.html:8]: decided not to keep it simple - here's what I've come up with:

Image
Image
Image

File: crt_portal/cts_forms/filters.py (hunk 1 above)
Comment (okkays) [reply crt_portal/cts_forms/filters.py:343]: I've updated the tooltips, plus the chips should help express this now, too

**Review by kewitham — APPROVED** [review]
lgtm!

**Pull request merged**